### ISSUE TITLE
generator copy errors out in py2.7.6

### ISSUE THREAD

**Issue by ejhumphrey** [pescadores/pescador#128]
*Note* this is almost certainly a wontfix, but I did want to document that I stepped on it.

I was attempting to use pescador in an Ubuntu 14.04 environment that's stuck at python 2.7.6 for reasons I can't help (so no lectures, plznthx) and stepped on an old Python `copy` bug: 

```python
"TypeError: object.__new__(generator) is not safe, use generator.__new__()"
```

This got fixed [once upon a time](https://bugs.python.org/issue11299), and is no longer an issue in the latest greatest python 2's. womp.

**Comment by bmcfee** [comment]
Strange.  Can you provide a code snippet to produce this behavior?

**Comment by ejhumphrey** [comment]
I'm 95% sure just iterating over a Streamer in 2.0 will cause it, based on the `copy.deepcopy` we're doing under the hood now.

on a related note: I *also* found that one way I'd been using pescador in 1.x is now broken in 2.x... tldr, all streams / generators must be wrapped in `pescador.Streamer` objects, otherwise the Streamer tries to copy a generator and fails in very unclear ways. I realize now this may be expected, and can open a separate issue if it seems relevant / we'd like to reproduce.

**Comment by cjacoby** [comment]
Correct; you can no longer wrap instantiated generators in Streamers, only
generator functions, because you can't copy generators with
`copy.deepcopy`. It is a somewhat unfortunate implication of the new
approach, but I think unavoidable.

I think a `PescadorError` subclass may be warranted for explaining what's
going on in this case, such that failing happens in a clear way.

**Comment by ejhumphrey** [comment]
Yea, I'd offer full throated support for providing more context for that error ... it was a headscratcher that I sorted kinda quickly on a lucky whim.
